src/js/player.js

```diff
@@ -4290,6 +4297,88 @@ class Player extends Component {
     return !!this.isAudio_;
   }
 
+  /**
+   * Get the current audioOnlyMode state or set audioOnlyMode to true or false.
+   *
+   * Setting this to `true` will hide all player components except the control bar,
+   * as well as control bar components needed only for video.
+   *
+   * @param {boolean} [value]
+   *         The value to set audioOnlyMode to.
+   *
+   * @return {boolean}
+   *         True if audioOnlyMode is on, false otherwise.
+   */
+  audioOnlyMode(value) {
+    if (value === undefined) {
+      return this.audioOnlyMode_;
+    }
+
+    if (value === this.audioOnlyMode_) {
+      return;
+    }
+
+    if (typeof value !== 'boolean') {
+      log.warn('audioOnlyMode can only be enabled or disabled using a boolean value');
+      return;
+    }
```

Comment:
I think this whole thing could be consolidated down to something similar to what's in #7629. And I'm not sure logging is particularly useful here. Maybe this:

```suggestion
    if (typeof value !== 'boolean' || value === this.audioOnlyMode_) {
      return this.audioOnlyMode_;
    }
```